### PR TITLE
[Documentation] Updated BLE API documentation and setup instructions

### PR DESCRIPTION
**Pull request by jimmy-huang**
This updates the API documentation for RSSI api and also added
a section for decribing client requirements and setup that's required
to test the BLE samples.

Signed-off-by: Jimmy Huang <jimmy.huang@intel.com>

### CODE REVIEW

File: docs/ble.md [hunk 1]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
 * `value` field with a string supplying the defined information
   * *NOTE: Values can also be Buffer objects, but that's not currently supported.*
 
+Client Requirements
+-------------------
+You can use any device that have BLE support to connect to the Arduino 101 when running
```
Comment (grgustaf) [inline docs/ble.md]: ha**s**

File: docs/ble.md [hunk 2]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
...
+ any of the BLE app or demos, we've succesfully tested the following setup:
```
Comment (grgustaf) [inline docs/ble.md]: app**s**, succes**s**fully
 "demos, we've" -> "demos. We've"

File: docs/ble.md [hunk 3]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
...
+
+* Update the bluetooth firmware, follow instructions [here](https://wiki.zephyrproject.org/view/Arduino_101#Bluetooth_firmware_for_the_Arduino_101)
```
Comment (grgustaf) [inline docs/ble.md]: **B**luetooth.

File: docs/ble.md [hunk 4]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
...
+* Any Android device running Android 6.0 Marshmallow or higher (We use Nexus 5/5X, iOS devices not tested)
+* Increase the X86 parition to 256K described [here](https://github.com/01org/zephyr.js#getting-more-space-on-your-arduino-101)
```
Comment (grgustaf) [inline docs/ble.md]: par**t**ition, described -> **as** described

File: docs/ble.md [hunk 5]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
...
+
+For the WebBluetooth Demo which support Physical Web, you'll need:
```
Comment (grgustaf) [inline docs/ble.md]: support**s the**

File: docs/ble.md [hunk 6]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
...
+* Chromium version 0.50 or higher
```
Comment (grgustaf) [inline docs/ble.md]: Do you mean version 50?

File: docs/ble.md [hunk 7]
```diff
@@ -202,7 +213,23 @@ The `init` object should contain:
...
+* Make sure Bluetooth on and Location Services enabled
```
Comment (grgustaf) [inline docs/ble.md]: **is** on

**Comment by jimmy-huang** [comment]
@grgustaf fixed typos :)

File: docs/ble.md [hunk 8]
```diff
@@ -215,16 +215,16 @@ The `init` object should contain:
...
 * Any Android device running Android 6.0 Marshmallow or higher (We use Nexus 5/5X, iOS devices not tested)
-* Increase the X86 parition to 256K described [here](https://github.com/01org/zephyr.js#getting-more-space-on-your-arduino-101)
+* Increase the X86 partition to 256K as described [here](https://github.com/01org/zephyr.js#getting-more-space-on-your-arduino-101)
 
-For the WebBluetooth Demo which support Physical Web, you'll need:
-* Chromium version 0.50 or higher
-* Make sure Bluetooth on and Location Services enabled
+For the WebBluetooth Demo which supports the Physical Web, you'll need:
+* Chromium version 50.0 or higher
+* Make sure Bluetooth on and Location Services is on
```
Comment (grgustaf) [inline docs/ble.md]: I meant **Bluetooth is on** instead of "Bluetooth on" - Location Services enabled was fine. :)

**Pull request merged**